rest/client.go

```diff
@@ -49,3 +54,63 @@ func (c *client) GetComponentsByAsset(ctx context.Context, id uuid.UUID, filter
 
 	return response.Components, nil
 }
+
+func (c *client) GetComponentRelations(ctx context.Context, id uuid.UUID, limit int, continuationToken string) (models.GetComponentRelationsResponse, error) {
+	request := rest.Get("/components/{component}/relations{?limit,continuation_token*}").
+		Assign("component", id).
+		Assign("limit", limit).
+		Assign("continuation_token", continuationToken).
+		SetHeader("Accept", "application/json")
+
+	var response models.GetComponentRelationsResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
+		return models.GetComponentRelationsResponse{}, err
+	}
+
+	return response, nil
+}
+
+func (c *client) GetRelatedComponents(ctx context.Context, id uuid.UUID, limit int, source, relationType, continuationToken string) (models.GetRelatedComponentsResponse, error) {
+	request := rest.Get("/relations/{source}/{type}/{id}/components{?limit,continuation_token*}").
+		Assign("source", source).
+		Assign("type", relationType).
+		Assign("id", id).
+		Assign("limit", limit).
+		Assign("continuation_token", continuationToken).
+		SetHeader("Accept", "application/json")
+
+	var response models.GetRelatedComponentsResponse
+	if err := c.DoAndUnmarshal(ctx, request, &response); err != nil {
+		return models.GetRelatedComponentsResponse{}, err
+	}
+
+	return response, nil
+}
+
+func (c *client) CreateComponentRelation(ctx context.Context, id uuid.UUID, relation models.Relation) error {
+	request := rest.Put("/components/{component}/relations").
+		Assign("component", id).
+		WithJSONPayload(relation).
+		SetHeader("Accept", "application/json")
+
+	if _, err := c.Do(ctx, request); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *client) DeleteComponentRelation(ctx context.Context, externalID, componentID uuid.UUID, source, relationType string) (err error) {
```

Comment:
This one should probably take a `models.Relation` as an argument instead of splitting the relation as arguments